packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -132,3 +154,224 @@ export const forwardFunds = async (
+    // throws if the pool has insufficient funds
+    borrower.borrow(tmpSeat, advanceAmount);
+
+    // this cannot throw since `.isSeen()` is called in the same turn
+    statusManager.advance(evidence);
+    const detail = {
+      txHash: evidence.txHash,
+      forwardingAddress: evidence.tx.forwardingAddress,
+      fullAmount,
+      destination,
+    };
+
+    try {
+      // XXX async flow types
+      await (localTransfer as unknown as GuestOf<typeof localTransfer>)(
+        tmpSeat,
+        poolAccount,
+        harden({ USDC: advanceAmount }),
+      );
+
+      // depositHandler.onfulFilled
+      tmpSeat.exit();
+      const amount = harden({
+        denom: usdc.denom,
+        value: advanceAmount.value,
+      });
+
+      const intermediateRecipient = getNobleICA().getAddress();
+
+      const destInfo = parseAccountIdArg(destination);
+      if (destInfo.namespace === 'cosmos') {
+        try {
+          await (destInfo.reference === settlementAddress.chainId
+            ? // local
+              poolAccount.send(destination, amount)
+            : // via IBC
+              poolAccount.transfer(destination, amount, {
+                forwardOpts: {
+                  intermediateRecipient,
+                },
+              }));
+          // transferHandler.onFulfilled
+          log('Advance succeeded', { advanceAmount, destination });
+          notifier.notifyAdvancingResult(detail, true);
+        } catch (error) {
+          await transferRejected(error);
+        }
+      } else if (supportsCctp(destination)) {
+        // send USDC via CCTP
+        try {
+          await poolAccount.transfer(intermediateRecipient, amount);
+          // transferCctpHandler.onFulfilled
+          // assets are on noble, transfer to dest.
+          const intermediaryAccount = getNobleICA();
+          try {
+            await intermediaryAccount.depositForBurn(destination, amount);
+            // transferHandler.onFulfilled
+            log('Advance succeeded', { advanceAmount, destination });
+            notifier.notifyAdvancingResult(detail, true);
+          } catch (error) {
+            await transferRejected(error);
+          }
+        } catch (error) {
+          transferCctpRejected(error);
+        }
+      } else {
+        // This is supposed to be caught in handleTransactionEvent()
+        Fail`🚨 can only transfer to Agoric addresses, via IBC, or via CCTP`;
+      }
+    } catch (error) {
+      depositRejected(error);
+    }
+
+    function depositRejected(error: any) {
+      // depositHandler.onRejected
+      log(
+        '⚠️ deposit to localOrchAccount failed, attempting to return payment to LP',
+        error,
+      );
+      try {
+        notifier.notifyAdvancingResult(detail, false);
+        borrower.returnToPool(tmpSeat, advanceAmount);
+        tmpSeat.exit();
+      } catch (e) {
+        log('🚨 deposit to localOrchAccount failure recovery failed', e);
+      }
+    }
```

Comment:
Any reason to define a `depositRejected` function here instead of inlining inside the `catch` ? Or do we prefer keeping same structure as original watch based logic.

Reply:
I started out inlining everything. But in one case, one handler was shared between 2 `watch(...)` calls, so that would have resulted in duplicating the handler. The `onFulfilled` part was ~2 lines, so I kept all the `onFulfilled` handlers inline and moved all the `onRejected` handlers to functions.

I could inline all the non-shared `onRejection` handlers, but I kinda like keeping the happy path concise.